spec/api-web-contents-spec.js

```diff
@@ -322,4 +323,208 @@ describe('webContents module', function () {
+        let zoomLevel = hostZoomMap[host]
+        if (!finalNavigation) {
+          w.webContents.setZoomLevel(zoomLevel)
+        }
+        e.sender.send(`${host}-zoom-set`)
+      })
+      ipcMain.on('host1-zoom-level', (e, zoomLevel) => {
+        let expectedZoomLevel = hostZoomMap['host1']
+        assert.equal(zoomLevel, expectedZoomLevel)
+        if (finalNavigation) {
+          done()
+        } else {
+          w.loadURL(`${zoomScheme}://host2`)
+        }
+      })
+      ipcMain.once('host2-zoom-level', (e, zoomLevel) => {
+        let expectedZoomLevel = hostZoomMap['host2']
+        assert.equal(zoomLevel, expectedZoomLevel)
+        finalNavigation = true
+        w.webContents.goBack()
+      })
+      w.loadURL(`${zoomScheme}://host1`)
+    })
+
+    it('can propagate zoom level across same session', (done) => {
+      const w2 = new BrowserWindow({
+        show: false
+      })
+      w2.webContents.on('did-finish-load', () => {
+        w.webContents.getZoomLevel((zoomLevel1) => {
+          assert.equal(zoomLevel1, hostZoomMap['host3'])
+          w2.webContents.getZoomLevel((zoomLevel2) => {
+            assert.equal(zoomLevel1, zoomLevel2)
+            w2.setClosable(true)
+            w2.close()
+            done()
+          })
+        })
+      })
+      w.webContents.on('did-finish-load', () => {
+        w.webContents.setZoomLevel(hostZoomMap['host3'])
+        w2.loadURL(`${zoomScheme}://host3`)
+      })
+      w.loadURL(`${zoomScheme}://host3`)
+    })
+
+    it('cannot propagate zoom level across different session', (done) => {
+      const w2 = new BrowserWindow({
+        show: false,
+        webPreferences: {
+          partition: 'temp'
+        }
+      })
+      let protocol = w2.webContents.session.protocol
+      protocol.registerStringProtocol(zoomScheme, (request, callback) => {
+        callback('hello')
+      }, (error) => {
+        if (error) return done(error)
+        w2.webContents.on('did-finish-load', () => {
+          w.webContents.getZoomLevel((zoomLevel1) => {
+            assert.equal(zoomLevel1, hostZoomMap['host3'])
+            w2.webContents.getZoomLevel((zoomLevel2) => {
+              assert.equal(zoomLevel2, 0)
+              assert.notEqual(zoomLevel1, zoomLevel2)
+              protocol.unregisterProtocol(zoomScheme, (error) => {
+                if (error) return done(error)
+                w2.setClosable(true)
+                w2.close()
+                done()
+              })
+            })
+          })
+        })
+        w.webContents.on('did-finish-load', () => {
+          w.webContents.setZoomLevel(hostZoomMap['host3'])
+          w2.loadURL(`${zoomScheme}://host3`)
+        })
+        w.loadURL(`${zoomScheme}://host3`)
+      })
+    })
+
+    it('can persist when it contains iframe', (done) => {
+      const server = http.createServer(function (req, res) {
+        setTimeout(() => {
+          res.end()
+        }, 2000)
```

Comment:
It was arbitrary, just wanted to make sure it doesn't become flaky. Have reduced it to 200, can increase if it turns flaky.